demo/actions.py

```diff
@@ -693,26 +716,37 @@ def run(self, dispatcher, tracker, domain):
             last_user_event = get_last_event_for(tracker, "user", skip=2)
             if last_user_event:
                 search_text = last_user_event.get("text")
+            else:
+                dispatcher.utter_message(text="Sorry, I can't answer your question.")
+                return []
 
         # Search of docs pages
         algolia = AlgoliaAPI(
             config.algolia_app_id, config.algolia_search_key, config.algolia_docs_index
         )
         alg_res = algolia.search(search_text)
 
-        doc_list = algolia.get_algolia_link(alg_res.get("hits"), 0)
-        doc_list += (
-            "\n" + algolia.get_algolia_link(alg_res.get("hits"), 1)
-            if len(alg_res.get("hits")) > 1
-            else ""
-        )
+        if alg_res and alg_res.get("hits") and len(alg_res.get("hits")) > 0:
+            doc_list = algolia.get_algolia_link(alg_res.get("hits"), 0)
+            doc_list += (
+                "\n" + algolia.get_algolia_link(alg_res.get("hits"), 1)
+                if len(alg_res.get("hits")) > 1
+                else ""
+            )
+
+            dispatcher.utter_message(
+                text="I can't answer your question directly, but I found the following from the docs:\n"
+                + doc_list
+            )
+            return [SlotSet("docs_found", True)]
 
         dispatcher.utter_message(
```

Comment:
I know it's the same logically, but can we put this in an `else:` to make it clearer off the bat?

Reply:
Do we want to actually say this though? i think we want it to be quiet about the fact that it didn't find stuff in the docs and go directly to the forum search as a first response instead. right?

Reply:
About the `else`, I will refactor it a little bit to get rid of the multiple return statements. That always makes it hard to read and dangerous to have logic fall through to the bottom without a proper return.

About the message when no docs are found. When I was testing it, I found it much better to provide consistent feedback, to avoid confusion.

If for some questions Sara starts talking about the docs, and for others only about the forum, as a user I am wondering why it is not telling me about the docs for some questions. It appears like a bug.

Reply:
Hm, interesting, I woudn't see it that way, but I guess that makes sense. You've been the one testing it, so I trust your UX gut feeling.